macOS/LocalPackages/NetworkProtectionMac/Sources/NetworkProtectionUI/VPNUIActionHandling/VPNUIActionHandling.swift

```diff
@@ -17,11 +17,18 @@
 //
 
 import Foundation
+import NetworkProtection
 
 public protocol VPNUIActionHandling {
     func moveAppToApplications() async
     func setExclusion(_ exclude: Bool, forDomain domain: String) async
     func shareFeedback() async
     func showPrivacyPro() async
     func showVPNLocations() async
+
+    /// Called when the user clicks on the toggle to stop the VPN.
+    ///
+    /// This allows overriding VPN tunnel control.
+    ///
+    func willStopVPN() async -> Bool
```

Comment:
Just offering a point where the parent app can override the VPN stop behaviour, and eventually even prevent it.

The idea is we show an alert when this is called and the user decides what to do.